### PR TITLE
chore(storage-browser): add destination text to upload controls view

### PR DESCRIPTION
**Pull request by timngyn**
#### Description of changes
- Add destination folder text to upload controls view
image

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx [hunk 1]
```diff
@@ -124,6 +128,7 @@ export const UploadControls = (): JSX.Element => {
       >
         Start upload
       </Primary>
+      <Destination>{history[history.length - 1]}</Destination>
```
Comment (timngyn) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx:131]: Should we only show the last folder or should we show the whole path for nested locations?

Currently this folder also includes a trailing slash as seen in the PR description. Should it be removed so it's only the folder name?

**Review by calebpollman — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx [hunk 2]
```diff
@@ -69,6 +70,15 @@ export const ActionIcon = ({ status }: ActionIconProps): React.JSX.Element => {
...
 };
 
+const Destination = ({ children }: { children?: React.ReactNode }) => {
+  return (
+    <DefinitionList className="storage-browser__destination">
+      <DefinitionTerm>Destination:</DefinitionTerm>
+      <DefinitionDetail>{children}</DefinitionDetail>
+    </DefinitionList>
+  );
+};
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx:80]: Don't think we need a list for a single element

**Review by calebpollman — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx (hunk 2 above)
Comment (hbuchel) [reply packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx]: Semantically it needs to be in a `dl` to use the other elements. It's fine to have a single definition.

Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx]: Got it

**Review by calebpollman — DISMISSED** [review]

File: packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx [hunk 3]
```diff
@@ -69,6 +70,17 @@ export const ActionIcon = ({ status }: ActionIconProps): React.JSX.Element => {
   );
 };
 
+const Destination = ({ children }: { children?: React.ReactNode }) => {
+  return (
+    <DefinitionList>
+      <DefinitionTerm className="storage-browser__destination">
```
Comment (hbuchel) [inline packages/react-storage/src/components/StorageBrowser/Views/LocationActionView/UploadControls.tsx:76]:
```suggestion
    <DefinitionList className="storage-browser__destination">
      <DefinitionTerm>
```

**Review by hbuchel — APPROVED** [review]

**Review by calebpollman — APPROVED** [review]

**Pull request merged**